### DISCUSSION TITLE
Unable to add unsupported device

### DISCUSSION THREAD

**Discussion by JRaeuber** [Koenkk/zigbee2mqtt#6306] [Devices]
Hello!

First of all, I am quite new to home assistant and I never had anything to do with javascprit, even if I do unterstand some basic programming.

I'm stuck at adding a new unsupported device.
It's an [IKEA Silverglans LED controller](https://www.ikea.com/de/de/p/silverglans-treiber-fuer-fernbedienung-dimmbar-weiss-10474772/).
Due to the compatibility to all the other IKEA devices I strongly assume that it works just the same way as the other two supported IKEA LED controller.
So my first attemp would be to just copy the configuration of one of the other IKEA LED controller from devices.js and change the needed parameters (name, model,..).

I have tried to add a new file (_mydevices.js_) to the zigbee2mqtt folder and add the filename below _external_converters:_ in the _configuration.yaml_ like written in the documentation.
I also tried to copy&paste the beginning section of the _devices.js_ with all the const definitions and so on.

All I get is a lot of errors in the log file starting with
```
internal/modules/cjs/loader.js:883
  throw err;
  ^
Error: Cannot find module './lib/ota'
Require stack:
- /app/lib/util/utils.js
- /app/lib/util/settings.js
- /app/index.js
```

Seems like I am doing something horrible wrong.

Could someone please tell me what I need to do to get my new light to work?

Thank you so much in advance!

Best regards
Jonas

**Comment by Koenkk** [comment]
- Can you provide the log after pairing the device?  Especially the line `Device '0x00158d0001dc126a' with Zigbee model 'XXXXX' and manufacturer name 'some_name' is NOT supported, please follow https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html`
- What features does this led controller support, only dimming or also color temperature?

**Reply by JRaeuber** [reply]
Thanks for your fast reply!

It says

`Device 'Bad_Spiegelschrank_Lampe' with Zigbee model 'SILVERGLANS IP44 LED driver' and manufacturer name 'IKEA of Sweden' is NOT supported, please follow https://www.zigbee2mqtt.io/how_tos/how_to_support_new_devices.html`

This LED controller only features dimming and I assume also all the other functions (transition, effects, moving/stepping,...) like the [already supported LED driver](https://www.zigbee2mqtt.io/devices/ICPSHC24-30EU-IL-1.html).

For me it seems like I just have to copy from the devices.js:

> zigbeeModel: ['TRADFRI transformer 30W', 'TRADFRI Driver 30W'],
>         model: 'ICPSHC24-30EU-IL-1',
>         vendor: 'IKEA',
>         description: 'TRADFRI driver for wireless control (30 watt)',
>         extend: preset.light_onoff_brightness(),
>         ota: ota.tradfri,
>         onEvent: ikea.bulbOnEvent,

modify it to something like this (with data from the label of the driver and what zigbee2mqtt tells me about it):

> zigbeeModel: ['SILVERGLANS IP44 LED driver'],
>         model: 'ICPSHC24-30-IL44-1',
>         vendor: 'IKEA',
>         description: 'IKEA SILVERGLANS IP44 LED driver for wireless control (30 watt)',
>         extend: preset.light_onoff_brightness(),
>         ota: ota.tradfri,
>         onEvent: ikea.bulbOnEvent,

and put it in my own devices file. 
But I guess there is something missing because without further information zigbee2mqtt doesn't know what to do with "preset.light_onoff_brightness()" for example.

**Reply by Koenkk** [reply]
Added, Changes will be available in the latest dev branch in a few hours (https://www.zigbee2mqtt.io/how_tos/how-to-switch-to-dev-branch.html)

**Reply by JRaeuber** [reply]
Thank you again!

But could you explain in a few words how I can add a device like this to my own device list?
Maybe not everyday when I get an unsupportet device someone add it for me.
Would be interesting to know how it works!

Best regards
Jonas

**Reply by Koenkk** [reply]
Can you post the content of your `mydevices.js` ?

**Reply by JRaeuber** [reply]
Right now it is empty, this LED driver would have been the first unsupported device for me.
I can't just add the part that you added above in the dev branch in my own device file, seems like there is missing some stuff around it - but which?

**Reply by Koenkk** [reply]
The syntax might differ a bit, you can find an example here: https://github.com/Koenkk/zigbee2mqtt.io/blob/master/docs/externalConvertersExample/light.js
